src/System.Net.Security/src/System/Net/SecureProtocols/SslStreamInternal.cs

```diff
@@ -225,26 +316,34 @@ private void ProcessWrite(byte[] buffer, int offset, int count)
             }
             finally
             {
-                _NestedWrite = 0;
+                if (asyncRequest == null || failed)
```

Comment:
(as an aside, this correctly handles the case similar to the one being fixed in #5551.)